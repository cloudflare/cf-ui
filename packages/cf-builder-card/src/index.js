import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
