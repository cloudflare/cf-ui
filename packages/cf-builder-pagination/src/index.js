import PaginationBuilder from './PaginationBuilder';

export { PaginationBuilder };
