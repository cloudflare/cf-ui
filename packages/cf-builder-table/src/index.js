import tableActions from './tableActions';
import TableActionTypes from './TableActionTypes';
import TableBuilder from './TableBuilder';
import TableBuilderPropTypes from './TableBuilderPropTypes';
import tableReducer from './tableReducer';

export {
  tableActions,
  TableActionTypes,
  TableBuilder,
  TableBuilderPropTypes,
  tableReducer
};
