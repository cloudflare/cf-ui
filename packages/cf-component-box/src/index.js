import { applyTheme } from 'cf-style-container';
import BoxUnstyled from './Box';
import BoxTheme from './BoxTheme';

const Box = applyTheme(BoxUnstyled, BoxTheme);

export { BoxUnstyled, Box, BoxTheme };
