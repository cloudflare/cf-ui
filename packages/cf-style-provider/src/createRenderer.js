import { createRenderer as createFelaRenderer } from 'fela';
import prefixer from 'fela-plugin-prefixer';
import fallbackValue from 'fela-plugin-fallback-value';
import unit from 'fela-plugin-unit';
import lvha from 'fela-plugin-lvha';
import validator from 'fela-plugin-validator';
import embedded from 'fela-plugin-embedded';
import beautifier from 'fela-beautifier';
import namedMediaQuery from 'fela-plugin-named-media-query';
import whitelistMediaQuery from './whitelistMediaQuery';
import { variables } from 'cf-style-const';

const defaultOpts = {
  dev: false
};

const mediaQueries = {
  mobile: `@media (min-width: ${variables.breakpoints.mobile})`,
  mobileWide: `@media (min-width: ${variables.breakpoints.mobileWide})`,
  tablet: `@media (min-width: ${variables.breakpoints.tablet})`,
  tabletLegacy: `@media (min-width: ${variables.breakpoints.tabletLegacy})`,
  tabletWide: `@media (min-width: ${variables.breakpoints.tabletWide})`,
  desktop: `@media (min-width: ${variables.breakpoints.desktop})`,
  desktopLarge: `@media (min-width: ${variables.breakpoints.desktopLarge})`,
  hover: `@media (hover)`
};

const removePrefix = query => query.replace('@media ', '');

const createRenderer = opts => {
  const usedOpts = Object.assign({}, defaultOpts, opts);
  const plugins = [
    prefixer(),
    fallbackValue(),
    unit(),
    lvha(),
    embedded(),
    namedMediaQuery(mediaQueries)
  ];
  const enhancers = [];

  if (usedOpts.dev === true) {
    plugins.push(validator());
    plugins.push(whitelistMediaQuery(mediaQueries));
    enhancers.push(beautifier());
  }

  return createFelaRenderer({
    plugins,
    enhancers,
    selectorPrefix: usedOpts.selectorPrefix,
    mediaQueryOrder: [
      removePrefix(mediaQueries.hover),
      removePrefix(mediaQueries.mobile),
      removePrefix(mediaQueries.mobileWide),
      removePrefix(mediaQueries.tablet),
      removePrefix(mediaQueries.tabletLegacy),
      removePrefix(mediaQueries.tabletWide),
      removePrefix(mediaQueries.desktop),
      removePrefix(mediaQueries.desktopLarge)
    ]
  });
};

export default createRenderer;
