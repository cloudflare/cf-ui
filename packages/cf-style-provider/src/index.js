import React, {
  Component,
  PropTypes,
  Children,
  isValidElement,
  cloneElement
} from 'react';
import { createRenderer as createFelaRenderer } from 'fela';
import prefixer from 'fela-plugin-prefixer';
import fallbackValue from 'fela-plugin-fallback-value';
import unit from 'fela-plugin-unit';
import lvha from 'fela-plugin-lvha';
import fontRenderer from 'fela-font-renderer';
import validator from 'fela-plugin-validator';
import beautifier from 'fela-beautifier';
import monolithic from 'fela-monolithic';
import { Provider } from 'react-fela';
import { variables } from 'cf-style-const';
import { ThemeProvider } from 'cf-style-container';

const defaultOpts = {
  selectorPrefix: 'cf-',
  dev: false,
  fontNode: undefined,
  cssNode: undefined
};

export const createRenderer = opts => {
  const usedOpts = Object.assign({}, defaultOpts, opts);
  const plugins = [prefixer(), fallbackValue(), unit(), lvha()];
  const enhancers = [fontRenderer(usedOpts.fontNode)];

  if (usedOpts.dev === true) {
    plugins.push(validator());
    enhancers.push(beautifier());
    enhancers.push(monolithic());
  }

  return createFelaRenderer({
    plugins,
    enhancers,
    selectorPrefix: [usedOpts.selectorPrefix]
  });
};

export const StyleProvider = ({
  selectorPrefix,
  dev,
  cssNode,
  fontNode,
  children,
  ...restProps
}) => {
  const renderer = createRenderer({
    selectorPrefix,
    dev,
    fontNode
  });
  const child = Children.only(children);
  return (
    <Provider renderer={renderer} mountNode={cssNode}>
      <ThemeProvider theme={variables}>
        {isValidElement(child) ? cloneElement(child, { ...restProps }) : child}
      </ThemeProvider>
    </Provider>
  );
};

StyleProvider.defaultProps = defaultOpts;
StyleProvider.propTypes = {
  dev: React.PropTypes.bool,
  selectorPrefix: React.PropTypes.string,
  cssNode: React.PropTypes.object,
  fontNode: React.PropTypes.object,
  children: PropTypes.node.isRequired
};
