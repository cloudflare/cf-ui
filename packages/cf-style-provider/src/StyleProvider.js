import React, {
  Component,
  Children,
  isValidElement,
  cloneElement
} from 'react';
import PropTypes from 'prop-types';
import createRenderer from './createRenderer';
import { Provider } from 'react-fela';
import { variables } from 'cf-style-const';
import { ThemeProvider } from 'cf-style-container';

const StyleProvider = ({ dev, children, renderer, ...restProps }) => {
  const providerRenderer =
    renderer ||
    createRenderer({
      dev
    });
  const child = Children.only(children);
  return (
    <Provider renderer={providerRenderer}>
      <ThemeProvider theme={variables}>
        {isValidElement(child) ? cloneElement(child, { ...restProps }) : child}
      </ThemeProvider>
    </Provider>
  );
};

StyleProvider.propTypes = {
  children: PropTypes.node,
  dev: PropTypes.bool,
  renderer: PropTypes.object
};

export default StyleProvider;
