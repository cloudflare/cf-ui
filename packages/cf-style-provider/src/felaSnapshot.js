import renderer from 'react-test-renderer';
import { html as beautify } from 'js-beautify';
import CleanCSS from 'clean-css';
import felaTestContext from './felaTestContext';
import createRenderer from './createRenderer';

const cssCleaner = new CleanCSS({ format: 'beautify', level: 0 });

export default component => {
  const felaRenderer = createRenderer({
    dev: true
  });

  return {
    component: renderer
      .create(felaTestContext(component, felaRenderer))
      .toJSON(),
    styles: cssCleaner.minify(felaRenderer.renderToString()).styles
  };
};
