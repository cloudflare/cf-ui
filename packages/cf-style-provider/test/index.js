import React from 'react';
import PropTypes from 'prop-types';
import { StyleProvider } from '../../cf-style-provider/src/index';
import { createComponent } from '../../cf-style-container/src/index';
import { Button } from '../../cf-component-button/src/index';
import renderer from 'react-test-renderer';
import { render } from 'react-dom';
import { mount } from 'enzyme';
import { html as beautify } from 'js-beautify';

test('StyleProvider should render', () => {
  const component = renderer.create(<StyleProvider><div /></StyleProvider>);
  expect(component.toJSON()).toMatchSnapshot();
});

test('StyleProvider should render styles', () => {
  const htmlNode = document.createElement('div');
  document.body.appendChild(htmlNode);

  const Foo = createComponent(() => ({
    margin: '10px',
    color: 'red'
  }));

  render(
    <StyleProvider>
      <Foo />
    </StyleProvider>,
    htmlNode
  );

  expect(
    beautify(document.documentElement.outerHTML, { indent_size: 2 })
  ).toMatchSnapshot();
});

test('StyleProvider should pass custom props through', () => {
  const Foo = () => <div />;
  const wrapper = mount(<StyleProvider number={10}><Foo /></StyleProvider>);
  wrapper.setProps({ something: true });
  expect(wrapper.prop('something')).toBeTruthy();
  expect(wrapper.prop('number')).toBe(10);
});
