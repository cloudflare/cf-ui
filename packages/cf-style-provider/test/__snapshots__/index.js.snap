// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleProvider should render 1`] = `<div />`;

exports[`StyleProvider should render styles 1`] = `
"<html>

<head>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\">
    .a {
      margin: 10px
    }

    .b {
      color: red
    }
  </style>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\" media=\\"(min-width: 64em)\\">
    .c {
      font-size: 1px
    }
  </style>
</head>

<body>
  <div>
    <div data-reactroot=\\"\\" class=\\"a b c\\"></div>
  </div>
</body>

</html>"
`;

exports[`felaSnapshot should return a formatted snapshot object 1`] = `
Object {
  "component": <div
    className="a"
/>,
  "styles": "
.a {
  color: black
}
",
}
`;
